### ISSUE TITLE
Permission error using Postgres16.1

### ISSUE THREAD

**Issue by slingao** [nautobot/nautobot#5103]
### Problem
Getting permission error in Postgres when running 'nautobot-server migrate' on new installation of nautobot.

### Environment
* Nautobot version (Docker tag too if applicable): 2.1.1
* Python version: 3.11.5
* Database platform, version: Postgres 16.1
* Middleware(s):

### Steps to Reproduce
1. Install Nautobot System Dependencies
2. Use Postgres 16.1
3. Run 'nautobot-server migrate'

I'm getting a permission error below:

$ nautobot-server migrate
Operations to perform:
  Apply all migrations: admin, auth, circuits, contenttypes, database, dcim, django_celery_beat, django_celery_results, extras, ipam, sessions, social_django, taggit, tenancy, users, virtualization
Running migrations:
Traceback (most recent call last):
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/utils.py", line 82, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.InsufficientPrivilege: permission denied for schema public
LINE 1: CREATE TABLE "django_migrations" ("id" bigserial NOT NULL PR...
                     ^

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/migrations/recorder.py", line 68, in ensure_schema
    editor.create_model(self.Migration)
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/base/schema.py", line 345, in create_model
    self.execute(sql, params or None)
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/base/schema.py", line 145, in execute
    cursor.execute(sql, params)
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/backends/utils.py", line 82, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.ProgrammingError: permission denied for schema public
LINE 1: CREATE TABLE "django_migrations" ("id" bigserial NOT NULL PR...
                     ^

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/nautobot/bin/nautobot-server", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/nautobot/lib64/python3.11/site-packages/nautobot/core/cli/__init__.py", line 52, in main
    run_app(
  File "/opt/nautobot/lib64/python3.11/site-packages/nautobot/core/runner/runner.py", line 297, in run_app
    management.execute_from_command_line([runner_name, command, *command_args])
  File "/opt/nautobot/lib64/python3.11/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/nautobot/lib64/python3.11/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/nautobot/lib64/python3.11/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/nautobot/lib64/python3.11/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/nautobot/lib64/python3.11/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/nautobot/lib64/python3.11/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
                         ^^^^^^^^^^^^^^^^^
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/migrations/executor.py", line 91, in migrate
    self.recorder.ensure_schema()
  File "/opt/nautobot/lib64/python3.11/site-packages/django/db/migrations/recorder.py", line 70, in ensure_schema
    raise MigrationSchemaMissing("Unable to create the django_migrations table (%s)" % exc)
django.db.migrations.exceptions.MigrationSchemaMissing: Unable to create the django_migrations table (permission denied for schema public
LINE 1: CREATE TABLE "django_migrations" ("id" bigserial NOT NULL PR...
                     ^
)

### Workaround
As workaround thanks to Jeffrey, added these extra commands in Postgres:

GRANT USAGE ON SCHEMA public TO nautobot;
ALTER DATABASE nautobot OWNER TO nautobot;

Reference: https://stackoverflow.com/questions/67276391/why-am-i-getting-a-permission-denied-error-for-schema-public-on-pgadmin-4

**Comment by glennmatthews** [comment]
Huh, I thought we'd addressed this in #4730 already for Postgres 15. I guess Postgres 16 changes the requirements a bit further?

**Comment by gsnider2195** [comment]
## Testing results

### Postgres 14

Using original schema (no `GRANT CREATE ON SCHEMA public TO nautobot;`)

```
postgres=# CREATE DATABASE nautobot;
CREATE DATABASE
postgres=# CREATE USER nautobot WITH PASSWORD 'secret';
CREATE ROLE
postgres=# GRANT ALL PRIVILEGES ON DATABASE nautobot TO nautobot;
GRANT
postgres=# \q
```

Migrations ran successfully

### Postgres 15

Add `GRANT CREATE ON SCHEMA public TO nautobot;` while connected to `postgres` database

```
postgres=# CREATE DATABASE nautobot;
CREATE DATABASE
postgres=# CREATE USER nautobot WITH PASSWORD 'secret';
CREATE ROLE
postgres=# GRANT ALL PRIVILEGES ON DATABASE nautobot TO nautobot;
GRANT
postgres=# GRANT CREATE ON SCHEMA public TO nautobot;
GRANT
postgres=# \q
```

Migrations fail:

```
    raise MigrationSchemaMissing("Unable to create the django_migrations table (%s)" % exc)
django.db.migrations.exceptions.MigrationSchemaMissing: Unable to create the django_migrations table (permission denied for schema public
LINE 1: CREATE TABLE "django_migrations" ("id" bigserial NOT NULL PR...
```

Add `GRANT CREATE ON SCHEMA public TO nautobot;` while connected to `nautobot` database

```
postgres=# \c nautobot
You are now connected to database "nautobot" as user "postgres".
nautobot=# GRANT CREATE ON SCHEMA public TO nautobot;
GRANT
```

Migrations ran successfully. Also confirmed that the `GRANT CREATE ON SCHEMA public TO nautobot;` on the `postgres` database was not required.

### Postgres 16

Schema from Postgres 15 used

```
postgres=# CREATE DATABASE nautobot;
CREATE DATABASE
postgres=# CREATE USER nautobot WITH PASSWORD 'secret';
CREATE ROLE
postgres=# GRANT ALL PRIVILEGES ON DATABASE nautobot TO nautobot;
GRANT
postgres=# \c nautobot
You are now connected to database "nautobot" as user "postgres".
nautobot=# GRANT CREATE ON SCHEMA public TO nautobot;
GRANT
nautobot=# \q

```

Migrations ran successfully. This confirms that #5175 will fix this issue.
